sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -228,4 +234,149 @@ public HttpPipeline getHttpPipeline() {
+    }
+
+    /**
+     * Gets a model, including the model metadata and the model definition.
+     * @param modelId The Id of the model.
+     * @return The ModelData and the http response
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<ModelData>> getModelWithResponse(String modelId) {
+        return withContext(context -> getModelWithResponse(modelId, context));
+    }
+
+    Mono<Response<ModelData>> getModelWithResponse(String modelId, Context context){
+        return protocolLayer
+            .getDigitalTwinModels()
+            .getByIdWithResponseAsync(modelId, includeModelDefinition, context);
+    }
+
+    /**
+     * Gets the list of models by iterating through a collection.
+     * @param listModelOptions The options to follow when listing the models. For example, the page size hint can be specified.
+     * @return A {@link PagedFlux} of ModelData and the http response.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<ModelData> listModels(ListModelOptions listModelOptions) {
+        return new PagedFlux<>(
+            () -> withContext(context -> listModelsSinglePageAsync(listModelOptions, context)),
+            nextLink -> withContext(context -> listModelsNextSinglePageAsync(nextLink, context)));
+    }
+
+    /**
+     * Gets the list of models by iterating through a collection.
+     * @return A {@link PagedFlux} of ModelData and the http response.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<ModelData> listModels() {
+        return listModels(new ListModelOptions());
+    }
+
+    PagedFlux<ModelData> listModels(Context context){
+        return new PagedFlux<>(
+            () -> listModelsSinglePageAsync(new ListModelOptions(), context),
+            nextLink -> listModelsNextSinglePageAsync(nextLink, context));
+    }
+
+    PagedFlux<ModelData> listModels(ListModelOptions listModelOptions, Context context){
+        return new PagedFlux<>(
+            () -> listModelsSinglePageAsync(listModelOptions, context),
+            nextLink -> listModelsNextSinglePageAsync(nextLink, context));
+    }
+
+    Mono<PagedResponse<ModelData>> listModelsSinglePageAsync(ListModelOptions listModelOptions, Context context){
+        return protocolLayer.getDigitalTwinModels().listSinglePageAsync(
+            listModelOptions.getDependenciesFor(),
+            listModelOptions.getIncludeModelDefinition(),
+            new DigitalTwinModelsListOptions().setMaxItemCount(listModelOptions.getMaxItemCount()),
+            context);
+    }
+
+    Mono<PagedResponse<ModelData>> listModelsNextSinglePageAsync(String nextLink, Context context){
+        return protocolLayer.getDigitalTwinModels().listNextSinglePageAsync(nextLink, context);
+    }
+
+    /**
+     * Deletes a model.
+     * @param modelId The id for the model. The id is globally unique and case sensitive.
+     * @return An empty Mono
```

Comment:
do we even need to mention this return?

Reply:
Yes, we will get doc verification errors.